### PR TITLE
Social distancing improvements doc

### LINKED ISSUE #14: Improve Social Distancing Code Documentation

### PR DESCRIPTION
**Pull request by mrn-mln**
Documenting functions, modules, classes, and methods plus rename variables and modules name based on PEP standard.
Tested on Coral Accelerator and EdgeTPU device.

### CODE REVIEW

File: applications/smart-distancing/libs/centroid_object_tracker.py [hunk 1]
```diff
@@ -7,77 +7,103 @@
 
 class CentroidTracker:
     """
-    a simple object tracker based on Euclidian distance of bounding boxes centroid of two consecutive frames.
-    if a box is losted betweeb two frames the tracker keep the box for next maxDisappeared frames.
-    """
-
-    def __init__(self, maxDisappeared=50):
+    A simple object tracker based on euclidean distance of bounding boxes centroid of two consecutive frames.
+    if a box is losted betweeb two frames the tracker keep the box for next max_disappeared frames.
```
Comment (rkgits) [inline applications/smart-distancing/libs/centroid_object_tracker.py]: typo: "if a box is lost between"
typo: " the tracker keeps"

File: applications/smart-distancing/libs/centroid_object_tracker.py [hunk 2]
```diff
@@ -7,77 +7,103 @@
 
 class CentroidTracker:
     """
-    a simple object tracker based on Euclidian distance of bounding boxes centroid of two consecutive frames.
-    if a box is losted betweeb two frames the tracker keep the box for next maxDisappeared frames.
-    """
-
-    def __init__(self, maxDisappeared=50):
+    A simple object tracker based on euclidean distance of bounding boxes centroid of two consecutive frames.
```
Comment (rkgits) [inline applications/smart-distancing/libs/centroid_object_tracker.py]: typo: "bounding box centroids"

File: applications/smart-distancing/libs/centroid_object_tracker.py (hunk 1 above)
Comment (rkgits) [inline applications/smart-distancing/libs/centroid_object_tracker.py]: Typo: "if box is lost between"
Typo: "the tracker keeps"

File: applications/smart-distancing/libs/centroid_object_tracker.py [hunk 3]
```diff
@@ -7,77 +7,103 @@
...
+    if a box is losted betweeb two frames the tracker keep the box for next max_disappeared frames.
 
-        """
-        maxDisappeared:if a box is losted betweeb two frames the tracker keep the box for next maxDisappeared frames.
+    :param max_disappeared: If a box is losted betweeb two frames the tracker keep the box for next
```
Comment (rkgits) [inline applications/smart-distancing/libs/centroid_object_tracker.py:13]: Typo: "if box is lost between"

Comment (rkgits) [reply applications/smart-distancing/libs/centroid_object_tracker.py:13]: Typo: "the tracker keeps"

File: applications/smart-distancing/libs/config_engine.py [hunk 4]
```diff
@@ -2,8 +2,17 @@
 import configparser
 import threading
 
+
 class ConfigEngine:
-    def __init__(self, config_path = './config-skeleton.ini'):
+    """
+    Handle the .ini confige file and provide a convenient interface to read the parameters from config.
+    When an instance of ConfigeEngine is created you can use or pass it to other classes/ modules in
+    which you want to access the parameters at config file
```
Comment (rkgits) [inline applications/smart-distancing/libs/config_engine.py]: Correction: " When an instance of ConfigeEngine is created you can use/pass it to other classes/modules that needs access to the parameters at config file. "

File: applications/smart-distancing/libs/detectors/jetson/mobilenet_ssd_v2.py [hunk 5]
```diff
@@ -1,25 +1,34 @@
-"""ssd.py
-This module implements the TrtSSD class.
-"""
 import ctypes
 import numpy as np
-import cv2
 import tensorrt as trt
-import pycuda.autoinit  # This is needed for initializing CUDA driver
 import pycuda.driver as cuda
+import pycuda.autoinit # This is needed for initializing CUDA driver
```
Comment (rkgits) [inline applications/smart-distancing/libs/detectors/jetson/mobilenet_ssd_v2.py]: Correction: "Required for initializing CUDA driver."

**Review by rkgits — COMMENTED** [review]
@mrn-mln really appreciate your contribution here. I reviewed the changes and I had some minor corrections on the description parts. Could you review my comments? 

Again, thanks for your collaboration.

File: applications/smart-distancing/libs/centroid_object_tracker.py [hunk 6]
```diff
@@ -7,77 +7,103 @@
...
+
+    def diregister(self, object_id):
+        """
+        Remove an object from objects and disappeared list.
 
-    def deregister(self, objectID):
-        del self.objects[objectID]
-        del self.disappeared[objectID]
+        Args:
+            object_id: A unique id for detected objects which is at objects list
```
Comment (mhejrati) [inline applications/smart-distancing/libs/centroid_object_tracker.py]: replace with "A unique id for detected objects which is at objects list" with "unique object ID"

File: applications/smart-distancing/libs/centroid_object_tracker.py [hunk 7]
```diff
@@ -7,77 +7,103 @@
...
+        """
+        del self.objects[object_id]
+        del self.disappeared[object_id]
 
     def update(self, object_list):
+        """
+        Updates the objects from the previous frame.
+        This function compares previous frame with current frame and take following actions:
+
+        1- Updates the bounding boxes of current frame, if finds the bounding boxs of the previous
```
Comment (mhejrati) [inline applications/smart-distancing/libs/centroid_object_tracker.py]: Replace with:
For each object that is being tracked, update its bounding box if it matches a detected object in the current frame.

File: applications/smart-distancing/libs/centroid_object_tracker.py [hunk 8]
```diff
@@ -7,77 +7,103 @@
...
+        frame are matched with current ones.
+        2- Registers an object as new one, if the current bounding boxes of that object is not
```
Comment (mhejrati) [inline applications/smart-distancing/libs/centroid_object_tracker.py]: Replace with:
For any detection in the current frame that doesn't match any tracked object, register it as a new object.

File: applications/smart-distancing/libs/centroid_object_tracker.py [hunk 9]
```diff
@@ -7,77 +7,103 @@
...
+        matched with any bounding boxes from previous frame.
+        3- Corresponds a bounding box as a lost bounding box and increments the counter of disappeared
```
Comment (mhejrati) [inline applications/smart-distancing/libs/centroid_object_tracker.py]: Replace with:
For each object that is being tracked and doesn't match a detection in the current frame, register it as lost and increment its disappeared counter.

File: applications/smart-distancing/libs/centroid_object_tracker.py [hunk 10]
```diff
@@ -7,77 +7,103 @@
...
+        object, if there is no matched bounding box from the previous frame with current frame's
+        bounding boxes.
+
+        Args:
+            object_list: A list of detected objects.
```
Comment (mhejrati) [inline applications/smart-distancing/libs/centroid_object_tracker.py]: Replace with:
object_list: List of detected objects.

File: applications/smart-distancing/libs/centroid_object_tracker.py [hunk 11]
```diff
@@ -7,77 +7,103 @@
...
+
+        Return:
+            objects: A list of updated objects.
```
Comment (mhejrati) [inline applications/smart-distancing/libs/centroid_object_tracker.py]: Replace with:
objects: List of updated objects.

File: applications/smart-distancing/libs/centroid_object_tracker.py [hunk 12]
```diff
@@ -7,77 +7,103 @@
...
+        """
         if len(object_list) == 0:
-            for objectID in list(self.disappeared.keys()):
-                self.disappeared[objectID] += 1
-                if self.disappeared[objectID] > self.maxDisappeared:
-                    self.deregister(objectID)
+            for object_id in list(self.disappeared.keys()):
+                self.disappeared[object_id] += 1
+                # Removed an object from the tracker when the object is not appear at max_disappeared perivous frames
```
Comment (mhejrati) [inline applications/smart-distancing/libs/centroid_object_tracker.py]: Replace with:
Removed an object from the tracker when the object is missing in the 'max_disappeared' previous frames.

File: applications/smart-distancing/libs/centroid_object_tracker.py (hunk 10 above)
Comment (mhejrati) [reply applications/smart-distancing/libs/centroid_object_tracker.py]: I suggest changing the variable name to detected_objects to clarify it.

File: applications/smart-distancing/libs/centroid_object_tracker.py (hunk 11 above)
Comment (mhejrati) [reply applications/smart-distancing/libs/centroid_object_tracker.py]: I suggest changing the variable name to tracked_objects to clarify it.

File: applications/smart-distancing/libs/centroid_object_tracker.py [hunk 13]
```diff
@@ -7,77 +7,103 @@
...
-                        self.deregister(objectID)
+                    if self.disappeared[object_id] > self.max_disappeared:
+                        self.diregister(object_id)
 
             else:
-                for col in unusedCols:
+                for col in unused_cols:
                     self.register(object_list[col])
 
         return self.objects
```
Comment (mhejrati) [inline applications/smart-distancing/libs/centroid_object_tracker.py]: TODO for future:
1- Refactor tracker logic, there is a lot of unnecessary casting and computation. 
2- Write unit tests.

File: applications/smart-distancing/libs/detectors/edgetpu/detector.py [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+        # Get model name from the config
+        self.name = self.config.get_section_dict('Detector')['Name']
+        if self.name == 'mobilenet_ssd_v2':  # or mobilenet_ssd_v1
+            from . import mobilenet_ssd
+            self.net = mobilenet_ssd.Detector(self.config)
+        else:
+            raise ValueError('Not supported network named: ', self.name)
+
+    def inference(self, resized_rgb_image):
+        # Here should inference on the image, output a list of objects, each obj is a dict with two keys "id" and "bbox" and "score"
```
Comment (mhejrati) [inline applications/smart-distancing/libs/detectors/edgetpu/detector.py]: Please replace with something like this:
"""
Run inference on an image.

input:

output:  a list of objects, each obj is a dict with two keys "id" and "bbox" and "score"

e.g. [{"id": 0, "bbox": [x1, y1, x2, y2], "score":s%}, {...}, {...}, ...]
"""

**Review by mhejrati — CHANGES_REQUESTED** [review]
@mrn-mln 
Thanks a lot for this PR. I just added some minor comments. Can you please apply and update the PR?

Many thanks.

File: applications/smart-distancing/libs/centroid_object_tracker.py (hunk 6 above)
Comment (mrn-mln) [reply applications/smart-distancing/libs/centroid_object_tracker.py:34]: Sure. Thank you for the comments.

**Comment by mrn-mln** [comment]
Thank you @mhejrati and @rk68 for the comments. I am updating the PR based on your feedback.

**Review by mhejrati — APPROVED** [review]

**Pull request merged**